### DISCUSSION TITLE
DIfference in number of multipliers while using bounded in opti

### DISCUSSION THREAD

**Discussion by sandeep026** [casadi/casadi#3419] [Q&A] [answered]
I am trying to minimize $x^2$ on the bounded interval $x \in [-10,-2]$, where the solution is $-2$.

I formulated the problem in opti using 2 ways for the constriants.

1.  using bounded:  ```bounded(-10,x,-2) ```
2.  using induvidual constraint ``` subject_to(x<=-2)``` and ```subject_to(-10<=x) ```

Why does opti report only a single multiplier when bounded is used  and what happens internally?

```
import casadi as cs

p1=cs.Opti()

x=p1.variable(1,1)


# Formulate the NLP
f = x**2


p1.minimize(f)

p1.subject_to(p1.bounded(-10,x,-2))

p1.solver('ipopt')

sol=p1.solve()

print('x:',sol.value(x))
print('lam_g',sol.value(p1.lam_g))

print(p1.lam_g.shape)


p2=cs.Opti()

x=p2.variable(1,1)

# Formulate the NLP
f = x**2 

p2.minimize(f)

p2.subject_to(-10<=x)
p2.subject_to(x<=-2)

p2.solver('ipopt')


sol=p2.solve()

print('x:',sol.value(x))
print('lam_g',sol.value(p2.lam_g))


```

**Comment by brunoml5** [comment]
Hi!

I did some [digging](https://github.com/casadi/casadi/blob/c0ec4276d3bd98212fd9a431ef707be93644ff21/casadi/core/optistack.hpp#L351), and it appears that when `bounded(lb, expr, ub)` is used, only one constraint is returned with `(lb<=expr)<= ub`. You can see from the ipopt output for `p1`:

```
Total number of inequality constraints...............:        1
        inequality constraints with only lower bounds:        0
   inequality constraints with lower and upper bounds:        1
        inequality constraints with only upper bounds:        0
```

Compared to the output for `p2`:

```
Total number of inequality constraints...............:        2
        inequality constraints with only lower bounds:        1
   inequality constraints with lower and upper bounds:        0
        inequality constraints with only upper bounds:        1
```

**Reply by sandeep026** [reply, marked as answer]
I should have closed this earlier. There is a neat explanation provided in [casadi ](https://optimization-online.org/wp-content/uploads/2018/01/6420.pdf)paper on page 18.
